Sources/_FoundationCShims/include/_CStdlib.h

```diff
@@ -60,7 +60,17 @@
 #endif
 
 #if __has_include(<signal.h>)
-#include <signal.h>
+/* wasi-libc's signal.h is available only if _WASI_EMULATED_SIGNAL is defined */
+# if !defined(__wasi__) || defined(_WASI_EMULATED_SIGNAL)
```

Comment:
One more small comment, could we use `TARGET_OS_WASI` here?

Also, if `_WASI_EMULATED_SIGNAL` isn't defined, what happens to `signal.h` on WASI? Does the file exist, it's just effectively empty and that's why the `__has_include` check passes?

Reply:
In wasi-libc, the `signal.h` header file itself is always available, but ff the macro isn't defined, it raises `#error` at compile-time: https://github.com/WebAssembly/wasi-libc/blob/b9ef79d7dbd47c6c5bafdae760823467c2f60b70/libc-top-half/musl/include/signal.h#L1-L3

The reason we need to conservative `_WASI_EMULATED_SIGNAL` guard here: We can assume all Foundation modules importing the CShim module defines `_WASI_EMULATED_SIGNAL`, so `defined(_WASI_EMULATED_SIGNAL)` shouldn't be false while building foundation. However, modules importing Foundation also transitively import the CShim and we can't assume `_WASI_EMULATED_SIGNAL` is defined while building those modules. So we need to be resilient to allow the both senario.

Reply:
Ah makes sense, thanks for the explanation!